src/audio/tdfb/tdfb.c

```diff
@@ -147,10 +176,116 @@ static int tdfb_init_coef(struct tdfb_comp_data *cd, int source_nch,
 		return -EINVAL;
 	}
 
-	coefp = ASSUME_ALIGNED(&config->data[0], 2);
+	if (config->num_angles > SOF_TDFB_MAX_ANGLES) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid num_angles %d",
+			    config->num_angles);
+		return -EINVAL;
+	}
+
+	if (config->beam_off_defined > 1) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid beam_off_defined %d",
+			    config->beam_off_defined);
+		return -EINVAL;
+	}
+
+	if (config->num_mic_locations > SOF_TDFB_MAX_MICROPHONES) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid num_mic_locations %d",
+			    config->num_mic_locations);
+		return -EINVAL;
+	}
+
+	/* In SOF v1.6 based beamformer topologies the legacy flag is set to indicate no
+	 * support for multiple angles, no mic locations, and no beam on/off switch.
+	 */
+	if (config->num_angles == 0 && config->num_mic_locations == 0) {
+		legacy = 1;
```

Comment:
do we know if we have any field deployments of this? If not I would argue we just return ABI not supported

Reply:
Yep, I'm not aware of any usage. Returning ABI not supported sounds like a good idea.